xcp_d/interfaces/connectivity.py

```diff
@@ -119,23 +140,321 @@ def _run_interface(self, runtime):
 
         masker = NiftiLabelsMasker(
             labels_img=atlas,
+            labels=node_labels,
             mask_img=mask,
             smoothing_fwhm=None,
             standardize=False,
+            resampling_target=None,  # they should be in the same space/resolution already
         )
 
         # Use nilearn for time_series
-        time_series = masker.fit_transform(filtered_file)
+        timeseries_arr = masker.fit_transform(filtered_file)
+        assert timeseries_arr.shape[1] == n_found_nodes
 
         # Apply the coverage mask
-        time_series[:, coverage_thresholded] = 0
+        timeseries_arr[:, coverage_thresholded] = np.nan
+
+        # Region indices in the atlas may not be sequential, so we map them to sequential ints.
+        seq_mapper = {idx: i for i, idx in enumerate(masker.labels_)}
+
+        if n_found_nodes != n_nodes:
+            # Fill in any missing nodes in the timeseries array with NaNs.
+            new_timeseries_arr = np.full(
+                (timeseries_arr.shape[0], n_nodes),
+                fill_value=np.nan,
+                dtype=timeseries_arr.dtype,
+            )
+            for col in range(timeseries_arr.shape[1]):
+                label_col = seq_mapper[masker.labels_[col]]
+                new_timeseries_arr[:, label_col] = timeseries_arr[:, col]
+
+            timeseries_arr = new_timeseries_arr
+
+            # Fill in any missing nodes in the coverage array with zero.
+            new_parcel_coverage = np.zeros(n_nodes, dtype=parcel_coverage.dtype)
+            for row in range(parcel_coverage.shape[0]):
+                label_row = seq_mapper[masker.labels_[row]]
+                new_parcel_coverage[label_row] = parcel_coverage[row]
+
+            parcel_coverage = new_parcel_coverage
+
+        # The time series file is tab-delimited, with node names included in the first row.
+        timeseries_df = pd.DataFrame(data=timeseries_arr, columns=node_labels)
+        correlations_df = timeseries_df.corr()
+        coverage_df = pd.DataFrame(data=parcel_coverage, index=node_labels)
+
+        timeseries_df.to_csv(self._results["timeseries"], sep="\t", na_rep="NaN", index=False)
```

Comment:
I started writing NaNs as "NaN" (instead of empty cells) in the tsv file. There might be something better (e.g., "n/a", as BIDS prescribes for missing data). Does anyone have a strong opinion on this?

Reply:
Actually switching to "n/a".